Updates to support EMC_verif-global major release

malloryprow opened the pull request:
EMC_verif-global's next major release version 2.0.0 needs to be used within the global workflow. This includes an upgrade to using MET version 9.1.3 and METplus version 3.1.1, as well as all the python scripts updated to using python3.

I tested on WCOSS Dell, Hera, and Orion (very low res single cycle), and the gfsmetp tasks all ran successfully.

malloryprow committed:
Add 'SDATE_GFS' rocoto env var for metp tasks

The connector script between EMC_verif-global and global-workflow
run_verif_global_in_global_workflow.sh (housed in EMC_verif-global)
contains a new feature that checks the maximum forecast hour for
verification is the hour's initialization date is equal to or after
the first GFS cycle initialization. If the date is before, the
maximum forecast hour is adjusted to the maximum hour relative to
the forecast verification date and the first GFS cycle initialization.
This feature was added to cut down on METplus calls being done
on data that don't exist.

Refs: #315

malloryprow committed:
Revamp of config.metp

Following the EMC_verif-global configuration remake, config.metp
has been revamped to support the new configuration settings.

Refs: #315

malloryprow committed:
Clean up and add new settings in metp.sh

SDATE_GFS was added into the script preamble following the
addition of the variable being added as a rocoto exported
environment variable for metp tasks.

Config.vrfy was removed from the list of relevant configs
to source.

Variables CDATEm1 and PDYm1 were removed and replaced with
VDATE. VDATE is the verification date to be done by the metp
tasks based on VRFYBACK_HRS set in config.metp.

Lines were added to catch the exit status of VERIFGLOBAL_SH
and to exit if not succesfully run so it can be caught by
rocoto.

Refs: #315

malloryprow committed:
Fix setting of 'SDATE_GFS' for free forecast XML

In the workflow XML for free forecasts there is no SDATE_GFS
variable. The 'SDATE_GFS' variable was changed to be set to the
value of 'SDATE'.

Refs: #315

malloryprow committed:
Update EMC_verif-global tag to verif_global_v2.0.0

RussTreadon-NOAA commented:
FYI, I'm tied up with v16.1, JEDI, and other tasks.  I won't be able to test or review the changes in this PR in the near future.  So as to not hold up this review are there any other staff you could invite to review this PR?

malloryprow commented:
@RussTreadon-NOAA Thanks for the heads up! GitHub listed you as a suggestion, and we have been corresponding a lot over METplus, so I followed the suggestion. I can add @yangfanglin as a reviewer instead. Not sure how tied up he is at the moment.

yangfanglin commented:
@Mallory Row - NOAA Affiliate ***@***.***> It is better for
someone else to review. I cannot find much time either.

On Wed, Apr 28, 2021 at 10:43 AM malloryprow ***@***.***>
wrote:

> @RussTreadon-NOAA <https://github.com/RussTreadon-NOAA> Thanks for the
> heads up! GitHub listed you as a suggestion, and we have been corresponding
> a lot over METplus, so I followed the suggestion. I can add @yangfanglin
> <https://github.com/yangfanglin> as a reviewer instead. Not sure how tied
> up he is at the moment.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/global-workflow/pull/317#issuecomment-828513133>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKY5N2J6AYX3BQENX3RQEJDTLANH5ANCNFSM43XFQZJA>
> .
>


-- 
*Fanglin Yang, Ph.D.*
*Chief, Model Physics Group*
*Modeling and Data Assimilation Branch*

*NOAA/NWS/NCEP Environmental Modeling Center*

*https://www.emc.ncep.noaa.gov/gmb/wx24fy/fyang/
<https://www.emc.ncep.noaa.gov/gmb/wx24fy/fyang/>*

malloryprow commented:
@yangfanglin Will do, thanks for letting me know!

malloryprow committed:
Fix typo in config.metp

There was a typo in the grid-to-obs METviewer related config variables.
The variables started with "g2g1" instead of "g2o1".

Refs: #315

malloryprow committed:
Update EMC_verif-global tag to verif_global_v2.0.1

Refs: #315

malloryprow committed:
Merge branch 'develop' into feature/EMC_verif-global_upgrade

malloryprow committed:
Merge branch 'develop' into feature/EMC_verif-global_upgrade

KateFriedman-NOAA commented:
@malloryprow I rolled the changes for this PR into a test of feature/hpc-stack (without waves) on Hera that ran from 2020090118 through 2020090500 for a C192C96L127 run. Please see my METplus output here on Hera:

```
-bash-4.2$ pwd
/scratch1/NCEPDEV/global/Kate.Friedman/archive/metplus_data
-bash-4.2$ ll by_VSDB/*/*/*/stacktest5/
```

Also, my EXPDIR with updated config.metp and ROTDIR with metp job logs are here:

EXPDIR: /scratch1/NCEPDEV/global/Kate.Friedman/expdir/stacktest5
ROTDIR: /scratch1/NCEPDEV/stmp4/Kate.Friedman/comrot/stacktest5
logs: /scratch1/NCEPDEV/stmp4/Kate.Friedman/comrot/stacktest5/logs/*/

Thanks!

malloryprow commented:
@KateFriedman-NOAA Output looks good!

KateFriedman-NOAA commented:
> @KateFriedman-NOAA Output looks good!

Great, thanks for checking @malloryprow ! I'm running the same experiment on WCOSS-Dell now and will share output for you to check when available. Thanks!

KateFriedman-NOAA commented:
@malloryprow I forgot to copy the updated config.metp into my EXPDIR on WCOSS-Dell and uncovered a bug in a verif-global script in the process.

Without the updated config the value of `$precip1_ccpa_accum24hr_fhr_max` didn't get set from the config and was set by the script default in `ush/run_verif_global_in_global_workflow.sh`:

`131 export precip1_ccpa_accum24hr_fhr_max=${precip1_ccpa_accum24hr_fhr_max:-$FHAX_GFS}`

However, since `$precip1_ccpa_accum24hr_fhr_max` isn't already set it tries to set it to the default which is `$FHAX_GFS`....I believe you meant that to be `$FHMAX_GFS` (the "M" is missing). Because of this the job doesn't get a value for that precip variable:

```
1376 + export precip1_ccpa_accum24hr_fhr_max=
1377 + precip1_ccpa_accum24hr_fhr_max=
```

...and fails with this error when the arguments for ndate are wrong:

```
+ precip1_ccpa_accum24hr_check_vhour=12
++ /gpfs/dell2/usrx/local/nceplibs/dev/hpc-stack/libs/hpc-stack/ips-18.0.1.163/prod_util/1.2.2/bin/ndate - 2020090200
ndate: Invalid option -
Usage: ndate [fhour [idate]]
+ precip1_ccpa_accum24hr_fhr_max_idate=
+ status=1
+ [[ 1 -ne 0 ]]
+ exit 1
```

See log on Mars: /gpfs/dell3/ptmp/Kate.Friedman/comrot/stack192/logs/2020090206/gfsmetpg2g1.log

The other two METplus jobs also failed from the same error:
/gpfs/dell3/ptmp/Kate.Friedman/comrot/stack192/logs/2020090206/gfsmetpg2g1.log
/gpfs/dell3/ptmp/Kate.Friedman/comrot/stack192/logs/2020090206/gfsmetppcp1.log

Thankfully an easy fix. Thanks! :)

malloryprow commented:
@KateFriedman-NOAA I sure did mean FHMAX_GFS haha. I'll do the bug fix for that, and then update the checkout.sh with the bug fix release.

malloryprow committed:
Update EMC_verif-global tag to verif_global_v2.0.2

Refs: #315

malloryprow commented:
@KateFriedman-NOAA checkout.sh tag updated!

KateFriedman-NOAA commented:
> @KateFriedman-NOAA checkout.sh tag updated!

Thanks for the quick fix @malloryprow ! I'm updating my Mars HOMEgfs sorc/verif-global.fd with the new tag, have copied the updated config.metp into my EXPDIR (what I failed to do earlier) and will rerun the failed jobs. Thanks!

KateFriedman-NOAA commented:
@malloryprow Reruns were successful. Please take a quick look at the following logs on Mars (that previously failed from that bug):

/gpfs/dell3/ptmp/Kate.Friedman/comrot/stack192/logs/2020090206/gfsmetp*.log
/gpfs/dell3/ptmp/Kate.Friedman/comrot/stack192/logs/2020090212/gfsmetp*.log

It's only the first two cycles that ran the METplus jobs so far so I didn't expect them to run long since there isn't much data yet but would be good to get early confirmation they did what was expected. My next cycles are now kicking off, I'll be running this test through 2020090500, as with the Hera run. Thanks!

malloryprow commented:
@KateFriedman-NOAA Yup, looking good so far!

KateFriedman-NOAA commented:
Thanks @malloryprow ! Will send more output/logs when the run completes.

malloryprow commented:
@KateFriedman-NOAA I checked the output on Mars in the metplus_data archive and all looks good!

KateFriedman-NOAA commented:
@malloryprow Excellent, thanks for checking and confirming!

KateFriedman-NOAA reviewed (changes requested):
Updates look good, just missing the verif-global tag update in Externals.cfg. Tested updates in 3 day run on both Hera and WCOSS-Dell. Please also update tag in Externals.cfg then this can be submitted. Thanks!

malloryprow committed:
Update Externals.cfg EMC_verif-global tag to verif_global_v2.0.2

Refs: #315

malloryprow commented:
Ah yes, forgot about that, done!

KateFriedman-NOAA reviewed (approved):
Thanks!

KateFriedman-NOAA commented:
Close #315
